include/xtensor/xpad.hpp

```diff
@@ -233,47 +233,80 @@ namespace xt
         return pad(e, pw, mode, constant_value);
     }
 
+    namespace detail {
+
+      template<class E>
+      auto resize(E & e, size_t n) -> decltype(e.resize(n), 1)
+      {
+        e.resize(n);
+        return n;
+      }
+
+      template<class E>
+      auto resize(E &, size_t n) -> decltype(std::tuple_size<E>::value)
+      {
+        XTENSOR_ASSERT(n == std::tuple_size<E>::value);
+        return n;
+      }
+
+      template <class E, class S>
+      inline auto tile(E&& e, const S& reps)
+      {
+          using size_type = typename std::decay_t<E>::size_type;
+          using value_type = typename std::decay_t<E>::value_type;
+          using return_type = temporary_type_t<value_type,
+                                               typename std::decay_t<E>::shape_type,
+                                               std::decay_t<E>::static_layout>;
+
+          XTENSOR_ASSERT(e.shape().size() == reps.size());
+
+          typename return_type::shape_type new_shape;
+          resize(new_shape, e.shape().size());
```

Comment:
You can actually use `xtl::make_sequence` to have it work with both `std::array` and vector-like containers:
```cpp
using new_shape_type = typename return_type::shape_type;
auto new_shape = xtl::make_sequence<new_shape_type>(e.shape().size());
```